quconot/implementations/mct_barenco_74_dirty.py

```diff
@@ -16,10 +16,10 @@
 
 class MCTBarenco74Dirty(MCTBase):
     def __init__(self, controls_no: int, **kwargs) -> None:
-        assert controls_no >= 2
+        if controls_no < 5:
+            raise ValueError("Number of controls must be >= 5 for this implementation")

```

Comment:
@ankit27kh is there a reason to change from assert to raise ValueError?

Reply:
But don't worry, I found out how to solve this. Using pytest for the raise error.

Reply:
@HandyKurniawan  this allows us more control
we should raise appropriate errors

Reply:
I agree with @ankit27kh, Assertion should be used only if something strange happened because of wrong implementation. Here it is user fault - so it should be `ValueError`